contracts/ProductNft/ProductNft.sol

```diff
@@ -0,0 +1,170 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
+import "@openzeppelin/contracts/utils/Counters.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+/**
+@title MembershipNft
+@author Marco Huberts & Javier Gonzalez
+@dev    Implementation of a Valorize Product Non Fungible Token using ERC721.
+*       Key information: the metadata should be ordered. The rarest NFTs should be the lowest tokenIds, then rarer and then rare NFTs.
+*/
+
+contract ProductNft is ERC1155 {
+    using Counters for Counters.Counter;
+
+    string public baseURI;
+    uint16 totalAmountOfTokenIds;
+    uint16 rarestTokensLeft;
+    uint16 rarerTokensLeft;
+    uint16 rareTokensLeft;
+    uint8 constant rarityRarest = 1;
+    uint8 constant rarityRarer = 2;
+    uint8 constant rarityRare = 3;
+    Counters.Counter public rarestTokenIds;
+    Counters.Counter public rarerTokenIds;
+    Counters.Counter public rareTokenIds;
+    uint256 public constant PRICE_PER_RAREST_TOKEN = 1.5 ether;
+    uint256 public constant PRICE_PER_RARER_TOKEN = 0.55 ether;
+    uint256 public constant PRICE_PER_RARE_TOKEN = 0.2 ether;
+    uint16 public startRarerTokenIdIndex;
+    uint16 public startRareTokenIdIndex;
```

Comment:
Let's move these 2 uint16 variables above, next to the other uint16s to do some struct packing. I would also move the baseURI to either the bottom, or at least above the uint256 storage variables.